### ISSUE TITLE
[Newbie] Some questions

### ISSUE THREAD

**Issue by lervag** [idanarye/vim-omnipytent#15]
I've created a task similar to this:

```python
@task
def new(ctx):
    """Create new issue from template."""
    number = FN.input('Insert new issue #: ')
    template = 'test/issues/template'
    newdir = f'test/issues/{number}'
    if not os.path.isdir(newdir):
        SH('cp', '-r', template, newdir)
        SH('git', 'add', '-f', newdir)
        CMD.edit(newdir + '/minivimrc')
```

This task works quite well; I use it to create new test files for vimtex issues. However, I would like to add some more verbosity. The problem is that the text is not displayed correctly. That is, if I add e.g. `print('newdir does not exist')` in an `else` branch, the text is _not_ displayed. The printing seems to work if I don't use the `FN.input`. So:

1. Is this a bug?
2. Or am I doing things in a bad way?

Should I perhaps use a different function/method to input the number argument? Note, I'm aware that I can but the number as a real argument here. In the end, I would like to do something like this:

```python
@task
def new(ctx, number=None):
    """Create new issue from template."""
    if number is None:
      number = FN.input('Insert new issue #: ')

    template = 'test/issues/template'
    newdir = f'test/issues/{number}'
    if os.path.isdir(newdir):
        print('Some warning message here')
    else:
        # Perhaps add a confirmation dialogue as well here.
        SH('cp', '-r', template, newdir)
        SH('git', 'add', '-f', newdir)
        CMD.edit(newdir + '/minivimrc')
```

**Comment by idanarye** [comment]
This may be related to Vim's quirkiness. `input()` keeps all the characters you have entered in the prompt - but swallows the `<Cr>`. There is no newline character after `input()`. You can try it yourself - run `:call input('') | echo 'AAA'`, type something, press `<Cr>` and see that "AAA" will be appended to the text you have typed, without separation.

The same goes with a `print()` from Python after `input()` - it doesn't create a new line. Since any normal person would expect a newline, you may not notice that the text was actually printed, on the same line as the `input` prompt.

If this is the case, try putting `print(' ')` before the text you actually want to print. It has to be a space (or any other non-empty string, but I guess you don't want to print anything visible) - an empty `print()` or even `print('')` won't work, and neither would `print('\n')`. Vim is quirky like that.

**Comment by lervag** [comment]
Thanks for the quick reply! I've noticed the same quirky behaviour, and I think I already tried what you suggest. But I might not have tried doing `print(' ')`, perhaps I only tried `print('')` and `print('\n')`. I'll test when I get home later today.

**Comment by lervag** [comment]
A minor follow-up-question: Is there an easy way to print with colors, or would I need to do `CMD.echohl` for that?

**Comment by lervag** [comment]
Ok, I've tested and investigated further. I've noticed that Omnipytent uses fzf by default if available, and I notice that when the task is run through fzf, then the echoing of messages does not work as expected.

Given the simple example:

```python
@task
def task(ctx):
    print('Line 1')
    print('Line 2')
```

If I do `:OP` (which I've mapped to `<leader>rr`), then by default, fzf is used to select the task, and for some reason, it either prints nothing, or it prints a single line. However, if I change to `ctrlp`, then both lines are printed with the "Press ENTER or type to return" message as expected.

I am not sure if this is a bug with Omnipytent, but it seems to me to be an unexpected side effect at the very least.

**Comment by lervag** [comment]
Btw, there seems to be a typo in the docs in line 1000, where `g:omnipytent_filePrefix` is used instead of `g:omnipytent_selectionUI`.

**Comment by idanarye** [comment]
Yea, this is a pain point for me as well... but since it bothers you too, I looked a bit into it and I may have a way around it.

The problem is that when you activate a function from an async event, multiple echoed lines will no longer "freeze" the UI. This also causes some problems with adding more async events, so I had some dirty workarounds to make multiple FZF launches work, but now I looked again and I think I can trick Vim using `feedkeys()` to have a more consistent behavior.
